x-pack/plugins/apm/common/agent_configuration/runtime_types/duration_rt.ts

```diff
@@ -6,22 +6,35 @@
 
 import * as t from 'io-ts';
 import { either } from 'fp-ts/lib/Either';
+import moment, { unitOfTime } from 'moment';
 import { amountAndUnitToObject } from '../amount_and_unit';
 
 export const DURATION_UNITS = ['ms', 's', 'm'];
 
-export function getDurationRt({ min }: { min: number }) {
+function getDuration({ amount, unit }: { amount: string; unit: string }) {
+  return moment.duration(parseInt(amount, 10), unit as unitOfTime.Base);
+}
+
+export function getDurationRt({ min, max }: { min?: string; max?: string }) {
   return new t.Type<string, string, unknown>(
     'durationRt',
     t.string.is,
     (input, context) => {
       return either.chain(t.string.validate(input, context), inputAsString => {
         const { amount, unit } = amountAndUnitToObject(inputAsString);
-        const amountAsInt = parseInt(amount, 10);
+        const inputDuration = getDuration({ amount, unit });
+        const minDuration = min
+          ? getDuration(amountAndUnitToObject(min))
```

Comment:
can you have a look at `parseEsInterval` from `src/plugins/data/common`? it's a little more strict and used in a lot more places. Not sure if it makes sense but we can give it a try.

Reply:
Unfortunately, we can't use it, since it throws an error for negative values and starting with 0. https://github.com/cauemarcondes/kibana/blob/agent-config-validations/src/plugins/data/common/search/aggs/date_interval_utils/parse_es_interval.test.ts#L57